crates/ra_analysis/src/completion/complete_keyword.rs

```diff
@@ -18,6 +18,16 @@ pub(super) fn complete_expr_keyword(acc: &mut Completions, ctx: &CompletionConte
     if !ctx.is_trivial_path {
         return;
     }
+
+    // complete keyword "crate" in use stmt
+    if let (Some(use_item), None) = (&ctx.use_item_syntax, &ctx.path_prefix) {
+        if use_item.use_tree().is_none() {
+            CompletionItem::new(CompletionKind::Keyword, "crate")
+                .kind(CompletionItemKind::Keyword)
```

Comment:
Let's add `::` automatically? To do this, I think you need to call `.snippet("crate::")`. And while we are at it, let's add `self` & `super` here as well?

Reply:
@matklad can i use `lookup` to filter the completion result. Now it strange to write tests when use lookup.

Reply:
I think the lookup should be just `crate`, and `::` should be handled by snippet. See how testing for `return;` works:

https://github.com/rust-analyzer/rust-analyzer/blob/fc2d7d1e1f03faa4566dea8b60eca860bcf9eb04/crates/ra_analysis/src/completion/complete_keyword.rs#L105